### PR TITLE
Feature/KMP Network Modules

### LINKED ISSUE #154: Create KMP Network Modules
## Overview (Required)

- For the Compose MultiPlatform Dodam Teacher, all core modules must be converted to KMP modules.

## Todo

- [ ] Migration to Android Native Module to KMP Module
- [ ] Create KMP Plugin
- [ ] Set Ktor iOS/Android Client

### PR DESCRIPTION
**Pull request by 8954sood**
## Overview (Required)

-  Originally, I planned to modify only the network layer in that branch, but there were too many modules that were linked, so I modified it all at once.
- Add Logging Module(with KmLog)
- I have switched Keystore to KMM module, but there is no keychain for iOS yet.
- Refactor Datastore Module to KMM module
- Refactor Common Module to KMM module
- Create Multiplatform plugins
- Refactor Network layer modules to KMM Module And Add Service Test Code

## Issue

- close #154

### CODE REVIEW

**Review by SeokgyuYun — APPROVED** [review]
LGTM 👍

File: build-logic/src/main/kotlin/com/b1nd/dodam/primitive/MultiplatformKoinPlugin.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.b1nd.dodam.dsl.library
+import com.b1nd.dodam.dsl.libs
+import com.b1nd.dodam.dsl.setupMultiplatform
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class MultiplatformKoinPlugin: Plugin<Project> {
+    override fun apply(target: Project) {
+        with(target) {
+
```
Comment (wsi1212) [inline build-logic/src/main/kotlin/com/b1nd/dodam/primitive/MultiplatformKoinPlugin.kt]: remove this empty line please

File: keystore/src/iosMain/kotlin/com/b1nd/dodam/keystore/KeyStoreManager.ios.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.b1nd.dodam.keystore
+
+actual class KeyStoreManager actual constructor() {
+    actual fun encrypt(plainText: String): String {
+        TODO("Not yet implemented")
```
Comment (wsi1212) [inline keystore/src/iosMain/kotlin/com/b1nd/dodam/keystore/KeyStoreManager.ios.kt:5]: please implementation this fun

**Review by wsi1212 — APPROVED** [review]
LGTM!! 😝

File: keystore/src/iosMain/kotlin/com/b1nd/dodam/keystore/KeyStoreManager.ios.kt (hunk 2 above)
Comment (8954sood) [reply keystore/src/iosMain/kotlin/com/b1nd/dodam/keystore/KeyStoreManager.ios.kt:5]: The iOS version of Keychain will be implemented in another branch.

**Pull request merged**